detox/src/devices/ios/AppleSimUtils.js

```diff
@@ -233,7 +233,12 @@ class AppleSimUtils {
   }
 
   recordVideo(udid, destination) {
-    return exec.spawnAndLog('/usr/bin/xcrun', ['simctl', 'io', udid, 'recordVideo', destination]);
+    const args = ['simctl', 'io', udid, 'recordVideo', destination];
+    if (codec) {
+      args.push('--codec');
+      args.push(codec);
```

Comment:
Pls test what happens if user passes a bad argument (like "vp9") here; will Detox catch a failure in the simctl command and complain to the user, or silently ignore this? The former sounds like a better option.

Reply:
It does fail and return an error code. Not sure if detox continues on afterwards or bails at that point. I'll have to test.
```
>> /usr/bin/xcrun simctl io booted recordVideo /tmp/video.mp4 -f --codec hsadf
(help text printed)
...
>> echo $?
117
```

Reply:
No worries, that should not be a problem. Detox will print a warning from stdout/stderr of `xcrun` .